modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js

```diff
@@ -1083,65 +1083,69 @@ class Sidebar extends Component {
 				id="accordion03"
 				role="tablist"
 			>
-				{group.map((key, index) => (
-					<div
-						class="panel panel-secondary"
-						key={`fields-group-${key}-${index}`}
-					>
-						<a
-							aria-controls="collapseTwo"
-							aria-expanded="true"
-							class="collapse-icon panel-header panel-header-link"
-							data-parent="#accordion03"
-							data-toggle="liferay-collapse"
-							href={`#ddm-field-types-${key}-body`}
-							id={`ddm-field-types-${key}-header`}
-							role="tab"
-						>
-							<span class="panel-title">
-								{fieldTypesGroup[key].label}
-							</span>
-							<span class="collapse-icon-closed">
-								<svg
-									aria-hidden="true"
-									class="lexicon-icon lexicon-icon-angle-right"
+				{group.map((key, index) => {
+					const fields = fieldTypesGroup[key].fields;
+
+					return (
+						fields.length > 0 && (
```

Comment:
Not sure if we settled on `!!fields.length &&` or this. I find this okay, but I think @wincent has raised this question in other occasion. 

I'm fine with the change, though. Maybe @diegonvs or @matuzalemsteles can take a quick look and forward from here if happy. Otherwise, we can send to forms.

Reply:
😮 I didn't know the "not false" operator (thanks 😄 ) I guess you are right, changing right away!

Reply:
> I think @wincent has raised this question in other occasion.

Probably, although I am too lazy to search for it now.

Basically, four pieces of relevant info:

- `!!` is a common idiom for coercing to boolean in JS (and other languages too).
- Since `0` is falsey in JS, it is "idiomatic" to write condtionals as `if (something.length)` instead of `if (something.length > 0)`; and due to falsiness of `0` we tend not to bother coercing in this case (eg. we don't do `if (!!something.length)`, although if we're writing TypeScript and we have a function that is annotated as returning a `boolean`, we may have to do something like `return !!something.length`).
- `a && b` evaluates to `a` (not necessarily `false`) when `a` is falsey.
- [In JSX `false` won't render anything at all](https://react-cn.github.io/react/tips/false-in-jsx.html) but `0` _will_ render, so while you might _want_ to write something like `fields.length && <div>blah</blah>` because it looks like "idiomatic" JS, in JSX it will actually squirt an unwanted `0` into the DOM, so, like @jbalsas said, you want to coerce to boolean with `!!fields.length &&`...

Reply:
Created https://github.com/liferay/liferay-frontend-projects/issues/384 to remind us to create some actual guidelines for this subject.